spec/models/event_spec.rb

```diff
@@ -290,4 +290,56 @@
       expect(event.participant_selection_locked).to eq(acceptances_sent || rejections_sent)
     end
   end
+
+  it "computes the email addresses of all participants" do
+    event = FactoryGirl.create(:event)
+    accepted_application_letter_1 = FactoryGirl.create(:application_letter_accepted, :event => event, :user => FactoryGirl.create(:user, :profile => FactoryGirl.create(:profile)))
+    accepted_application_letter_2 = FactoryGirl.create(:application_letter_accepted, :event => event, :user => FactoryGirl.create(:user, :profile => FactoryGirl.create(:profile)))
+    accepted_application_letter_3 = FactoryGirl.create(:application_letter_accepted, :event => event, :user => FactoryGirl.create(:user, :profile => FactoryGirl.create(:profile)))
+    rejected_application_letter = FactoryGirl.create(:application_letter_rejected, :event => event, :user => FactoryGirl.create(:user))
+    [accepted_application_letter_1, accepted_application_letter_2, accepted_application_letter_3, rejected_application_letter].each { |letter| event.application_letters.push(letter) }
+    expect(event.email_addresses_of_participants(true, [], [])).to include(accepted_application_letter_1.user.email, accepted_application_letter_2.user.email, accepted_application_letter_3.user.email)
+  end
+
+  it "computes the email addresses of a group" do
+    event = FactoryGirl.create(:event)
+    accepted_application_letter_1 = FactoryGirl.create(:application_letter_accepted, :event => event, :user => FactoryGirl.create(:user, :profile => FactoryGirl.create(:profile)))
+    accepted_application_letter_2 = FactoryGirl.create(:application_letter_accepted, :event => event, :user => FactoryGirl.create(:user, :profile => FactoryGirl.create(:profile)))
+    accepted_application_letter_3 = FactoryGirl.create(:application_letter_accepted, :event => event, :user => FactoryGirl.create(:user, :profile => FactoryGirl.create(:profile)))
+    rejected_application_letter = FactoryGirl.create(:application_letter_rejected, :event => event, :user => FactoryGirl.create(:user))
+    participant_group1 = FactoryGirl.create(:participant_group, :event => event, :user => accepted_application_letter_1.user, :group => 2)
+    [accepted_application_letter_1, accepted_application_letter_2, accepted_application_letter_3, rejected_application_letter].each { |letter| event.application_letters.push(letter) }
+    result = event.email_addresses_of_participants(false, [participant_group1.group], [])
+    expect(result).to include(accepted_application_letter_1.user.email)
+    expect(result).to_not include(accepted_application_letter_2.user.email, accepted_application_letter_3.user.email)
+  end
+
+
+  it "computes the email addresses of certain participants" do
```

Comment:
Please also test what happens when both `groups` and `users` are non-empty in email_addresses_of_participants